pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -775,6 +776,11 @@ func (cb *ClusterBuilder) buildUpstreamClusterTLSContext(opts *buildClusterOpts,
 				CaCertificatePath: tls.CaCertificates,
 			}
 
+			// Use service accounts in service if TLS settings does not have subject alt names for service entries.
```

Comment:
I am a bit confused why we don't need this for the L765 branch?

Reply:
I did not know that credential name uses the subject alt names. I think it needs to be added